src/mSupplyMobileApp.js

```diff
@@ -83,21 +83,27 @@ class MSupplyMobileAppContainer extends React.Component {
     this.handleBackEvent = this.handleBackEvent.bind(this);
     this.getCanNavigateBack = this.getCanNavigateBack.bind(this);
     this.renderPageTitle = this.renderPageTitle.bind(this);
-    this.scheduler.schedule(this.synchronise,
-                            SYNC_INTERVAL);
+    this.scheduler.schedule(this.synchronise, SYNC_INTERVAL);
     this.scheduler.schedule(() => {
-      if (this.state.currentUser !== null) { // Only reauthenticate if currently logged in
+      if (this.state.currentUser !== null) {
+        // Only reauthenticate if currently logged in
         this.userAuthenticator.reauthenticate(this.onAuthentication);
       }
     }, AUTHENTICATION_INTERVAL);
   }
 
   componentDidMount() {
     BackHandler.addEventListener('hardwareBackPress', this.handleBackEvent);
+    if (this.state.isInitialised && this.synchroniser.isSyncing()) {
```

Comment:
Just did a quick test with a console log in `MSupplyMobileAppContainer` In `constructor`, `componentWillMount`, `componentDidMount` and `componentWillUnmount`.

First 3 only called on app start (putting in background doesn't seem to trigger any). Closing app triggers unmount. Apparently at that point this.database.removeAllListeners() is undefined anyway!

Plot gets deeper a little: Closing the app with back button triggers unmount, but going to the android open app manager view and swiping the app away _doesn't_ trigger the unmount code.